### PR TITLE
Added docstrings for protected functions in regression metrics

### LINKED ISSUE #389: Add docstrings to protected functions
## 📚 Documentation

Add docstrings to all protected functions at least tell what are the arguments for...
This can see easily implemented in multiple PRs :rabbit:

### PR DESCRIPTION
**Pull request by DevPranjal**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements): Yes, issue #389 
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section? Yes :)
- [x] Did you make sure to update the docs? This PR updates the docs
- [ ] Did you write any new necessary tests? No

## What does this PR do?
Partialy #389 

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃
Yes, a lot... Along with great learnings

### CODE REVIEW

File: torchmetrics/functional/regression/cosine_similarity.py [hunk 1]
```diff
@@ -31,6 +40,25 @@ def _cosine_similarity_update(
...
+        preds: Predicted tensor
+        target: Ground truth tensor
+        reduction:
+            The method of reducing along the batch dimension using sum, mean or taking the individual scores
+
+    Example:
+        >>> target = torch.tensor([[1, 2, 3, 4],
+        ...                        [1, 2, 3, 4]])
+        >>> preds = torch.tensor([[1, 2, 3, 4],
+        ...                       [-1, -2, -3, -4]])
```
Comment (Borda) [inline torchmetrics/functional/regression/cosine_similarity.py:56]:
```suggestion
        >>> target = torch.tensor([[1, 2, 3, 4], [1, 2, 3, 4]])
        >>> preds = torch.tensor([[1, 2, 3, 4], [-1, -2, -3, -4]])
```

File: torchmetrics/functional/regression/explained_variance.py [hunk 2]
```diff
@@ -41,6 +50,36 @@ def _explained_variance_compute(
...
+        >>> target = torch.tensor([[0.5, 1], [-1, 1], [7, -6]])
+        >>> preds = torch.tensor([[0, 2], [-1, 2], [8, -5]])
+        >>> n_obs, sum_error, sum_squared_error, sum_target, sum_squared_target = _explained_variance_update(preds,
+        ...                                                                                                  target)
+        >>> _explained_variance_compute(n_obs,
+        ...                             sum_error,
+        ...                             sum_squared_error,
+        ...                             sum_target,
+        ...                             sum_squared_target,
+        ...                             multioutput='raw_values')
```
Comment (Borda) [inline torchmetrics/functional/regression/explained_variance.py:79]:
```suggestion
        >>> n_obs, sum_error, sum_squared_error, sum_target, sum_squared_target = \
        ...     _explained_variance_update(preds, target)
        >>> _explained_variance_compute(
        ...     n_obs, sum_error, sum_squared_error, sum_target, sum_squared_target, multioutput='raw_values')
```

**Review by Borda — APPROVED** [review]

File: torchmetrics/functional/regression/r2.py [hunk 3]
```diff
@@ -21,6 +21,15 @@
 
 
 def _r2_score_update(preds: Tensor, target: Tensor) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
```
Comment (DevPranjal) [inline torchmetrics/functional/regression/r2.py:23]: I'm a bit skeptical about the variable names defined in this function. For example, `sum_error = torch.sum(target, dim=0)`. Hence the docstring for `_r2_score_compute` is currently incomplete.

**Review by SkafteNicki — APPROVED** [review]

File: torchmetrics/functional/regression/r2.py (hunk 3 above)
Comment (Borda) [reply torchmetrics/functional/regression/r2.py:23]: hi, how would you suggest to change it?

Comment (DevPranjal) [reply torchmetrics/functional/regression/r2.py:23]: Currently it is as follows,
```
sum_error = torch.sum(target, dim=0)
sum_squared_error = torch.sum(target * target, dim=0)
diff = target - preds
residual = torch.sum(diff * diff, dim=0)
total = target.size(0)
```
Suggested,
```
sum_obs = torch.sum(target, dim=0)
sum_squared_obs = torch.sum(target * target, dim=0)
residual = target - preds
rss = torch.sum(residual * residual, dim=0)
n_obs = target.size(0)
```
And similar changes in the argument names for `_r2_score_compute()`

Comment (Borda) [reply torchmetrics/functional/regression/r2.py:23]: yep, that sounds good to me :rabbit:

Comment (DevPranjal) [reply torchmetrics/functional/regression/r2.py:23]: I have made the required changes.
